code/core/src/types/modules/core-common.ts

```diff
@@ -211,13 +210,37 @@ export type Options = LoadOptions &
   CLIOptions &
   BuilderOptions & { build?: TestBuildConfig };
 
+// A minimal version of Polka's interface to avoid exposing internal implementation details
+export type Middleware<T extends IncomingMessage = IncomingMessage> = (
+  req: T & IncomingMessage,
+  res: ServerResponse,
+  next: (err?: string | Error) => Promise<void> | void
+) => Promise<void> | void;
+
+interface ServerApp<T extends IncomingMessage = IncomingMessage> {
+  server: HttpServer;
+
+  use(pattern: RegExp | string, ...handlers: Middleware<T>[]): this;
+  use(...handlers: Middleware<T>[]): this;
+
+  get(...handlers: Middleware<T>[]): this;
+  post(...handlers: Middleware<T>[]): this;
+  put(...handlers: Middleware<T>[]): this;
+  patch(...handlers: Middleware<T>[]): this;
+  delete(...handlers: Middleware<T>[]): this;
+  head(...handlers: Middleware<T>[]): this;
+  options(...handlers: Middleware<T>[]): this;
+  connect(...handlers: Middleware<T>[]): this;
+  trace(...handlers: Middleware<T>[]): this;
+}
+
 export interface Builder<Config, BuilderStats extends Stats = Stats> {
   getConfig: (options: Options) => Promise<Config>;
   start: (args: {
     options: Options;
     startTime: ReturnType<typeof process.hrtime>;
-    router: Router;
-    server: Server;
+    router: ServerApp;
+    server: HttpServer;
```

Comment:
Are these not the same thing now?

Reply:
not quite. `server` is (and has always been) the underlying server from the `node:http` module, where `router` is the `polka` app that wraps it.
It is true that there is no difference now between the `router` and the "`app`", but they are not the same as the `server`.